raiden/tests/integration/test_blockchainservice.py

```diff
@@ -257,8 +259,13 @@ def test_blockchain(
     assert channel_manager_address == event['channel_manager_address'].decode('hex')
     assert token_proxy.address == event['asset_address'].decode('hex')
 
+    contract_data = solidity_get_contract_data(
+        registry_contracts,
+        get_contract_path('ChannelManagerContract.sol'),
+        'ChannelManagerContract'
+    )
     channel_manager_proxy = jsonrpc_client.new_contract_proxy(
-        registry_contracts['ChannelManagerContract']['abi'],
+        contract_data['abi'],
```

Comment:
`compile_file` just compiles the file and returns the output. 
`solidity_get_contract_data()` is a new function I added in the [linked](https://github.com/ethereum/pyethapp/pull/188) pyethapp PR, to abstract away the reading of the contract data in a backwards compatible way which would work with all current `solc` versions.